tool/plugin/lib/edit.dart

```diff
@@ -4,14 +4,271 @@
  * found in the LICENSE file.
  */
 
-//import 'dart:async';
-//import 'dart:convert';
-//import 'dart:io';
+import 'dart:async';
+import 'dart:io';
 
-//import 'package:path/path.dart' as p;
+import 'build_spec.dart';
+import 'util.dart';
 
-//import 'build_spec.dart';
-//import 'globals.dart';
+Set<EditCommand> appliedEditCommands = {};
 
-class Edit {
-}
\ No newline at end of file
+void checkAndClearAppliedEditCommands() {
+  if (appliedEditCommands.length != editCommands.length) {
+    var commands = <EditCommand>{};
+    commands.addAll(editCommands);
+    commands.removeAll(appliedEditCommands);
+    separator("UNUSED EditCommand");
+    commands.forEach((cmd) => log(cmd.toString()));
+  }
+  appliedEditCommands.clear();
+}
+
+List<EditCommand> editCommands = [
+  EditAndroidModuleLibraryManager(),
+  EditFlutterProjectSystem(),
+  EditFlutterDescriptionProvider(),
+  Subst(
+      path: 'flutter-studio/src/io/flutter/project/FlutterProjectSystem.java',
+      initial:
+          'gradleProjectSystem.getAndroidFacetsWithPackageName(project, packageName, scope)',
+      replacement: 'Collections.emptyList()',
+      version: '4.0'),
+  Subst(
+      path:
+          'flutter-studio/src/io/flutter/actions/FlutterShowStructureSettingsAction.java',
+      initial: 'import com.android.tools.idea.gradle.structure.actions.AndroidShowStructureSettingsAction;',
+      replacement: 'import com.android.tools.idea.gradle.actions.AndroidShowStructureSettingsAction;',
+      versions: ['3.6', '4.0', '2020.1']),
+  Subst(
+      path: 'flutter-studio/src/io/flutter/actions/OpenAndroidModule.java',
+      initial: 'findGradleTarget',
+      replacement: 'findImportTarget',
+      versions: ['3.6', '4.0', '2020.1']),
+  Subst(
+      path: 'src/io/flutter/utils/AndroidUtils.java',
+      initial:
+          'import com.android.tools.idea.gradle.dsl.parser.BuildModelContext;',
+      replacement:
+          'import com.android.tools.idea.gradle.dsl.model.BuildModelContext;',
+      version: '4.1'),
```

Comment:
this might format better if you end the version param w/ a trailing comma: `version: '4.1',),`

Reply:
Yes, it looks a little better, thanks.